tock/tock/static/js/components/notifications.js

```diff
@@ -0,0 +1,50 @@
+(function setup_notifications() {
+  if("Notification" in window) {
+    Notification.requestPermission().then(function(result) {
```

Comment:
Looks like it's captured just inside the beginning with this:

```javascript
// If the user grants permission, or has granted permission in the past,
// setup the timers.  Otherwise bail out.
if (result !== 'granted') {
    return;
}
```